- Added spin state order handling to PolarizationCorrectionFredrikze Correction Algorithm (#37672)
mantidproject/mantid: docs/source/algorithms/PolarizationCorrectionFredrikze-v1.rst

> To update your documentation based on the new approach of using spin states directly rather than flipper configurations, you need to revise the relevant sections to reflect this change. Here’s an updated version of your documentation with a focus on the new spin state configurations and removing references to flipper configurations:
> 
> ---

Reviewer: We should highlight things like this in the release notes rather than at the top of the docs, which have a standardised style.
```suggestion
```

Description
-----------

Performs wavelength polarization correction on a TOF reflectometer spectrometer.

> The algorithm is based on the paper Fredrikze, H, et al. "Calibration of a polarized neutron reflectometer" Physica B 297 (2001).

Reviewer: Could we add an explicit reference here? (The doc for `DepolarizedAnalyserTransmission.rst` has an example of how to do this.

Polarizer and Analyzer efficiencies are calculated and used to perform an intensity correction on the input workspace. The input workspace(s) are in units of wavelength (inverse angstroms).

In the ideal case :math:`P_{p} = P_{a} = A_{p} = A_{a} = 1`.

:math:`\rho = \frac{P_{a}}{P_{p}}` where rho is bounded by, but inclusive of 0 and 1. Since this ratio is wavelength dependent, rho is a polynomial, which is expressed as a function of wavelength. For example:
:math:`\rho(\lambda) =\sum\limits_{i=0}^{i=2} K_{i}\centerdot\lambda^i`, can be provided as :math:`K_{0}, K_{1}, K_{2}`.

:math:`\alpha = \frac{A_{a}}{A_{p}}` where alpha is bounded by, but inclusive of 0 and 1. Since this ratio is wavelength dependent, alpha is a polynomial, which is expressed as a function of wavelength. For example:
:math:`\alpha(\lambda) =\sum\limits_{i=0}^{i=2} K_{i}\centerdot\lambda^i`, can be provided as :math:`K_{0}, K_{1}, K_{2}`.

Spin State Configurations
#########################

> The algorithm directly uses spin states to define the measurement setup. The spin states are specified using two properties:

Reviewer: ```suggestion
The algorithm expresses the order of the workspaces in the input and output groups in terms of spin states. These orders are specified using two properties:
```

**Input Spin State Order:**
This property determines the order of the spin states in the input workspace group

- :literal:`'pp, pa, ap, aa'`
   Full polarization corrections: both parallel, parallel-anti-parallel, anti-parallel-parallel, both  anti-parallel. Four input workspaces are required. The spin states can be provided in any order and should match the order of the workspaces in the input group.

- :literal:`'p, a'`

>    PNR polarization corrections:  parallel, anti-parallel. Two input workspaces are required. The spin states can be provided in any order and should match the order of the workspaces in the input group.

Reviewer: Can we make clear here that `PA` is Full polarization corrections and what `PNR` stands for?

.. note::
    the default order used by the algorithm if none is specified are :literal:`pp, pa, ap, aa` for full polarization analysis or :literal:`p, a` for PNR.

**Output Spin State Order:**
This property determines the order of the spin states in the output workspace group. Similar to the input configuration, users can specify output spin states in any order.

> Output from Full Polarization Analysis

Reviewer: Can we have an extra heading for the outputs from the PNR version?

4               :math:`I_{aa}`
==============  ================

Output from Polarized Neutron Reflectivity
------------------------------------------

> The output of this algorithm is a :ref:`WorkspaceGroup <WorkspaceGroup>`, the resulting :ref:`WorkspaceGroup <WorkspaceGroup>` will have 2 entries, corresponding to the output specified spin state orders, if no specific output spin states are specified, it will be formatted as follows:

Reviewer: ```suggestion
The output of this algorithm is a :ref:`WorkspaceGroup <WorkspaceGroup>`. The resulting :ref:`WorkspaceGroup <WorkspaceGroup>` will have 2 entries, corresponding to the specified output spin state orders. If no specific output spin states are specified, it will be formatted as follows:
```

==============  ================
Entry in group  Measurement
==============  ================
1               :math:`I_{p}`
2               :math:`I_{a}`
==============  ================

Usage
-----

.. include:: ../usagedata-note.txt

**Example - PolarizationCorrectionFredrikze (PNR)**

.. testcode:: PolarizationCorrectionFredrikze (PNR)

    # import mantid algorithms, numpy and matplotlib
    from mantid.simpleapi import *
    import matplotlib.pyplot as plt
    import numpy as np

    run = Load("POLREF00004699.nxs")
    eff = Load("fredrikze_efficiencies_37125.nxs")
    run = ConvertUnits(run, "Wavelength")
    eff = ConvertUnits(eff, "Wavelength")
    eff = RebinToWorkspace(eff, run.getItem(0), True)

>     out = PolarizationCorrectionFredrikze(run, eff, "PNR", InputSpinStates="pp, pa, ap, aa", OutputSpinStates="p, a")

Reviewer: Does a PNR example typically take 4 workspaces an input or should it only take two?

> 

Reviewer: There should probable be a `.. testoutput::` here as in the one from the wildes docs. That way we can make sure this usage example consistently performs correctly.

4               :math:`I_{aa}`
==============  ================

This format allows for detailed analysis of the polarization states.

.. categories::

.. sourcelink::

---

> 

Reviewer: Is there a reason for this extra line being included?
